contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,453 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.10;
+
+import {IERC20} from "../interfaces/IERC20.sol";
+import {IsOHM} from "../interfaces/IsOHM.sol";
+import {IgOHM} from "../interfaces/IgOHM.sol";
+import {IStaking} from "../interfaces/IStaking.sol";
+import {SafeERC20} from "../libraries/SafeERC20.sol";
+import {YieldSplitter} from "../types/YieldSplitter.sol";
+import {OlympusAccessControlled, IOlympusAuthority} from "../types/OlympusAccessControlled.sol";
+
+/**
+    @title  YieldDirectorV2 (codename Tyche) 
+    @notice This contract allows donors to deposit their gOHM and donate their rebases
+            to any address. Donors will be able to withdraw the sOHM equivalent of their principal
+            gOHM at any time. Donation recipients can also redeem accrued rebases at any time.
+    @dev    Any functions dealing with initial deposits will take an address (because no ID has been
+            assigned). After a user has deposited, all functions dealing with deposits (like
+            withdraw or redeem functions) will take the ID of the deposit. All functions that return
+            aggregated data grouped by user will take an address (iterates across all relevant IDs).
+ */
+contract YieldDirectorV2 is YieldSplitter, OlympusAccessControlled {
+    using SafeERC20 for IERC20;
+
+    error YieldDirector_InvalidDeposit();
+    error YieldDirector_InvalidUpdate();
+    error YieldDirector_InvalidWithdrawal();
+    error YieldDirector_NotYourDeposit();
+    error YieldDirector_NoDeposits();
+    error YieldDirector_NoRedeemableBalance();
+    error YieldDirector_WithdrawalsDisabled();
+    error YieldDirector_RedeemsDisabled();
+
+    address public immutable sOHM;
+    IStaking public immutable staking;
+    mapping(address => uint256[]) public recipientIds;
```

Comment:
similar here, comment like this would be nice for out of context readers.
address -> Array of the deposit id's user is recipient of